src/domain/NewNotificationRepository.ts

```diff
@@ -14,6 +15,58 @@ export type NewNotificationRecord = {
   output: Response<202, NewNotificationResponse> | Response<403, UnauthorizedMessageBody>;
 };
 
+export const isNewNotificationRecord = (record: AllRecord): O.Option<NewNotificationRecord> =>
+  record.type === 'NewNotificationRecord' ? O.some(record) : O.none;
+
+export const hasSuccessfulResponse = (record: NewNotificationRecord) => record.output.statusCode === 202;
+
+export const hasRecipientTaxId = (record: NewNotificationRecord) =>
+  pipe(
+    record.input.body.recipients,
+    RA.every((recipient) => pipe(recipient.taxId, O.fromNullable, O.isSome))
+  );
+
+export const hasRecipientDigitalDomicile = (record: NewNotificationRecord) =>
+  pipe(
+    record.input.body.recipients,
+    RA.every((recipient) => pipe(recipient.digitalDomicile, O.fromNullable, O.isSome))
+  );
+
+export const hasPhysicalAddress = (record: NewNotificationRecord) =>
+  pipe(
+    record.input.body.recipients,
+    RA.every((recipient) => pipe(recipient.physicalAddress, O.fromNullable, O.isSome))
+  );
+
+export const hasRegisteredLetterAsPhysicalDocumentType = (record: NewNotificationRecord) =>
+  record.input.body.physicalCommunicationType === PhysicalCommunicationTypeEnum.REGISTERED_LETTER_890;
+
+export const hasRecipientPaymentCreditorTaxId = (record: NewNotificationRecord) =>
+  pipe(
+    record.input.body.recipients,
+    RA.every((recipient) =>
+      pipe(
+        recipient.payment,
+        O.fromNullable,
+        O.map((_) => _.creditorTaxId),
+        O.isSome
+      )
+    )
```

Comment:
What do you think about this suggestion? Feel free to keep it or close it

```suggestion
    RA.every(({ payment }) => pipe(payment?.creditorTaxId, O.fromNullable, O.isSome))
```

Reply:
Nice! Done! 🍻